### PR TITLE
fix: improve handling of pseudo classes and elements

### LINKED ISSUE #52: Empty-sub-selector warning while using pseudo selectors in :has/:where/:is
This may be an issue for all angular apps since it's using critters/beasties under the hood for inlining critical styles. It seems that parsing css that contains pseudo selectors into :has/:where log this warning.

While rendering ssr apps, this may pollute logs since inlining may be done for every page request

```css
    .ui-form-control:where(:not([readonly])):where(:active, :focus, :focus-visible, [data-focused]) {
    }
    [role=group]:has(> .ui-form-control):has(.ui-form-control:where([disabled], [data-disabled])) {
    }
    .ui-form-control:where([aria-invalid=true], :invalid, [data-invalid=true]) {
    }
```

Error:

```
2 rules skipped due to selector errors:
  .ui-form-control:where(:not([readonly])):where( , , [data-focused]) -> Empty sub-selector
  .ui-form-control:where([aria-invalid=true], , [data-invalid=true]) -> Empty sub-selector
Merging inline stylesheets into a single <style> tag skipped, no inline stylesheets to merge
```

Maybe related to this 
- https://github.com/GoogleChromeLabs/critters/issues/103#issuecomment-2222879997
- https://github.com/GoogleChromeLabs/critters/issues/153
- https://github.com/angular/angular-cli/issues/25725

### PR DESCRIPTION
**Pull request by alan-agius4**
Closes #52

<details open>
<summary>:sparkles: <i><h3>Description by Callstackai</h3></i></summary>

This PR improves the handling of pseudo classes and elements in the Beasties library by updating the regex patterns used for normalization.

<details>
<summary><b>Diagrams of code changes</b></summary>

```mermaid
sequenceDiagram
    participant Test
    participant Beasties
    participant CSS Parser

    Test->>Beasties: process(HTML with stylesheet)
    Note over Beasties: New pattern added for<br/>pseudo-classes/elements

    Beasties->>CSS Parser: Parse CSS from style.css
    CSS Parser-->>Beasties: Return CSS rules

    Note over Beasties: Remove pseudo-classes/elements<br/>from selectors during normalization
    
    Beasties->>Beasties: normalizedSelector()<br/>Remove :pseudo-classes<br/>Remove ::pseudo-elements
    
    Beasties-->>Test: Return processed HTML with:<br/>- Original pseudo selectors preserved<br/>- Inlined styles in head
```

</details>

<details>
<summary><b>Files Changed</b></summary>
<table>
<tr><th>File</th><th>Summary</th></tr>
<tr><td><a href=https://github.com/danielroe/beasties/pull/58/files#diff-4235a696ef2eaf34ccb14e2449cf3ef61d531df3e462f6ffe11c4b516f280f25>packages/beasties/src/index.ts</a></td><td>Updated regex pattern to handle pseudo classes and elements.</td></tr>
<tr><td><a href=https://github.com/danielroe/beasties/pull/58/files#diff-849e319a3a7f331b4d494ee9ca7d1e87800e7cfe48ec30be99f440f809225397>packages/beasties/test/__snapshots__/beasties.test.ts.snap</a></td><td>Added snapshot for new test case involving pseudo classes and elements.</td></tr>
<tr><td><a href=https://github.com/danielroe/beasties/pull/58/files#diff-85d39184055aba70288d5466be440921fe7bc8cda59fece2a61ae7dcb03c2b87>packages/beasties/test/beasties.test.ts</a></td><td>Added a test case to verify the handling of pseudo classes and elements.</td></tr>

</table>
</details>

*This PR includes files in programming languages that we currently do not support. We have not reviewed files with the extensions `.ts`, `.snap`. <a href=https://docs.callstack.ai/introduction>See list of supported languages.</a>*

</details>

## Summary by CodeRabbit

- **New Features**
	- Enhanced CSS selector processing to support both pseudo-classes and pseudo-elements.
	- Improved normalization of CSS selectors for better stylesheet handling.

- **Tests**
	- Introduced a new test case to verify handling of CSS pseudo-classes and pseudo-elements, ensuring correct processing and logger behaviour.

### CODE REVIEW

File: packages/beasties/src/index.ts [hunk 1]
```diff
@@ -26,8 +26,7 @@ import { applyMarkedSelectors, markOnly, parseStylesheet, serializeStylesheet, v
 import { createDocument, serializeDocument } from './dom'
 import { createLogger, isSubpath } from './util'
 
-const removePseudoElementsPattern = /(?<!\\)::?[a-z-]+(?![a-z-(])/gi
-const cleanEmptyNotPattern = /::?not\(\s*\)/g
```
Comment (danielroe) [inline packages/beasties/src/index.ts:30]: do we not need to handle an empty `::not()`?

Comment (alan-agius4) [reply packages/beasties/src/index.ts:30]: It's also handled by the updated RegExp

```
/(?<!\\)::?[a-z-]+(?:\(.+\))?/gi.test(`p:not(.irrelevant)`)
Outputs: true
```

Comment (danielroe) [reply packages/beasties/src/index.ts:30]: no... curious we were handling it before - maybe it might end up empty based on stripping of unused selectors?

Comment (alan-agius4) [reply packages/beasties/src/index.ts:30]: Stripping happens at a later stage, and I do not see any failing tests 🤷‍♂️

**Review by danielroe — APPROVED** [review]

**Pull request merged**